server/modules/syllabus/database/syllabus-repository.ts

```diff
@@ -37,4 +37,13 @@ export class SyllabusRepository
     }
     return document
   }
+
+  async updateSyllabusFilteredByOwner (id: string, owner: string, data: Partial<Syllabus>): Promise<Syllabus | null> {
+    const document = await this.EntityModel.findOneAndUpdate({ _id: id, owners: owner }, data, { new: true })
+
+    if (document) {
+      return this.mapper.toDomainEntity(document)
+    }
+    return document
```

Comment:
Same here. I prefer to manage errors with the `if` instead of managing the correct path.

```suggestion
    if (!document) {
      return null
    }
    return this.mapper.toDomainEntity(document)
```